bluemira/base/components.py

```diff
@@ -89,7 +89,11 @@ def __repr__(self) -> str:
         """
         return self.name + " (" + self.__class__.__name__ + ")"
 
-    def filter_components(self, names: Iterable[str]):
+    def filter_components(
+        self,
+        names: Iterable[str],
+        filter_: Optional[Callable[[PhysicalComponent], bool]] = None,
```

Comment:
Because filter is a builtin: https://docs.python.org/3/library/functions.html?highlight=isinstance#filter its also the same variable name that anytree uses